lib/python/treadmill_aws/ec2client.py

```diff
@@ -86,9 +97,40 @@ def list_instances(ec2_conn, ids=None, tags=None, hostnames=None, state=None):
 
     for reservation in reservations:
         instances.extend(reservation['Instances'])
+
+    # working around a bug in instance-lifetime filter
+    if spot is True:
+        return list(filter(lambda r: 'InstanceLifecycle' in r, instances))
+    elif spot is False:
+        return list(filter(lambda r: 'InstanceLifecycle' not in r, instances))
     return instances
 
 
+def list_spot_requests(ec2_conn):
+    """Returns a list of object-like spot instance requests
+        # sir.status:
+        # open       The request is waiting to be fulfilled.
+        # active     The request is fulfilled and has an associated Instance
+        # failed     The request has one or more bad parameters.
+        # closed     The Spot Instance was interrupted or terminated.
+        # cancelled  You cancelled the request, or the request expired.
+    """
+    requests = []
+    sirs = ec2_conn.describe_spot_instance_requests()['SpotInstanceRequests']
+    for _ in sirs:
```

Comment:
Refactored. I generally prefer to use a struct like SimpleNamespace over dictionaries, I find them more readable, less typing and more natural to use that dictionaries. For this same reason, I am a bit disappointed that the whole treadmill_aws project is using the boto3 basic client instead of the "Service Resource" client which returns AWS objects instead of huge dictionaries.